server/channels/store/sqlstore/utils.go

```diff
@@ -196,3 +196,27 @@ func quoteColumnName(driver string, columnName string) string {
 
 	return columnName
 }
+
+// scanRowsIntoMap scans SQL rows into a map, using a provided scanner function to extract key-value pairs
+func scanRowsIntoMap[K comparable, V any](rows *sql.Rows, scanner func(rows *sql.Rows) (K, V, error), defaults map[K]V) (map[K]V, error) {
+	results := make(map[K]V)
```

Comment:
Same here -- if we have a defaults, we know the minimum cap needed. Not a huge win, I know.